tests/api-watch-diff/test.ml

```diff
@@ -1,4 +1,428 @@
+open Types
+
 let%expect_test "test_diff_interface" =
   let result = Api_watch_diff.diff_interface ~reference:[] ~current:[] in
   Format.printf "%b" result;
   [%expect {|false|}]
+
+let ref_signature =
+  (* Signature for ref.mli:
+     > type t = int
+     > type unused_type = string
+     > val f t-> string *)
+  [
+    Sig_type
+      ( Ident.create_persistent "t",
+        {
+          type_params = [];
+          type_arity = 0;
+          type_kind = Type_abstract;
+          type_private = Public;
+          type_manifest =
+            Some
+              (Transient_expr.type_expr
+                 (Transient_expr.create (Tvar (Some "int")) ~level:0 ~scope:0
```

Comment:
I suspect one problem here (perhaps the main problem) is that you are creating this as a `TVar` which I think is a type variable like `'a`: 

https://github.com/ocaml/ocaml/blob/303f6e2eb13d3b45341a9ba3e1e971f8df7ca685/typing/types.mli#L65-L67

I think what we actually want is a `Tconstr`:

https://github.com/ocaml/ocaml/blob/303f6e2eb13d3b45341a9ba3e1e971f8df7ca685/typing/types.mli#L79-L81

I think you can see an example of how to construct the integer and float types here

https://github.com/ocaml/ocaml/blob/303f6e2eb13d3b45341a9ba3e1e971f8df7ca685/typing/predef.ml#L67-L70

I wonder if we could/should try to use the `Predef` module in these tests... WDYT, @NathanReb?

Reply:
I'm not extremely familiar with these parts of the compiler yet. I think a good way to better understand this is to figure out a way to pretty print the content of `.cmi` files.

The compiler as several IR (intermediate representation) and most of them can be pretty printed with the `-d<IR>` flags. E.g. you can get the compiler to pretty print the parsetree for a given file with `-dparsetree`. I couldn't find one that would print the right thing but that's something to look into.

If it doesn't exist we can also simply write our own using `[@@deriving show]` or `[@@deriving pp]`.

Anyway, this is a task for later, I'll look deeper into the `Predef` module and how `.cmi` files are compiled and report back!